Source/DafnyLanguageServer/MarkupTestFile.cs

```diff
@@ -0,0 +1,183 @@
+  /// that needs to be provided are strings that encode every bit of state necessary in the string
+  /// itself.
+  /// 
+  /// The current set of encoded features we support are: 
+  /// 
+  /// $$ - The position in the file.  There can be at most one of these.
+  /// 
+  /// [| ... |] - A span of text in the file.  There can be many of these and they can be nested
+  /// and/or overlap the $ position.
+  /// 
+  /// {|Name: ... |} A span of text in the file annotated with an identifier.  There can be many of
+  /// these, including ones with the same name.
+  /// 
+  /// Additional encoded features can be added on a case by case basis.
+  /// </summary>
+  public static class MarkupTestFile {
+    private const string SpanStartString = "[>";
+    private const string SpanEndString = "<]";
+    private const string NamedSpanEndString = "<}";
+
+    private static void Parse(
+        string input, out string output, out List<int> positions, out IDictionary<string, List<TextSpan>> spans) {
+      positions = new List<int>();
+      var tempSpans = new Dictionary<string, List<TextSpan>>();
+
+      var outputBuilder = new StringBuilder();
+
+      // A stack of span starts along with their associated annotation name.  [||] spans simply
+      // have empty string for their annotation name.
+      var spanStartStack = new Stack<(int matchIndex, string name)>();
+      var namedSpanStartStack = new Stack<(int matchIndex, string name)>();
+
+      var r = new Regex(@"(?<Position>\>\<)|(?<SpanStart>\[\>)|(?<SpanEnd>\<\])|(?<NameSpanStart>\{\>([-_.A-Za-z0-9\+]+)\:)|(?<NameSpanEnd>\<\})");
+      var outputIndex = 0;
+      var inputIndex = 0;
+      foreach (Match match in r.Matches(input)) {
+        var diff = inputIndex - outputIndex;
+        var matchIndexInOutput = match.Index - diff;
+        var outputPart = input.Substring(inputIndex, match.Index - inputIndex);
+        outputIndex += outputPart.Length;
+        outputBuilder.Append(outputPart);
+        inputIndex = match.Index + match.Length;
+        if (match.Groups["Position"].Success) {
+          positions.Add(matchIndexInOutput);
+        } else if (match.Groups["SpanStart"].Success) {
+          spanStartStack.Push((matchIndexInOutput, string.Empty));
+        } else if (match.Groups["NameSpanStart"].Success) {
+          namedSpanStartStack.Push((matchIndexInOutput, string.Empty));
+        } else if (match.Groups["SpanEnd"].Success) {
+          PopSpan(spanStartStack, tempSpans, matchIndexInOutput);
+        } else if (match.Groups["NameSpanEnd"].Success) {
+          PopSpan(namedSpanStartStack, tempSpans, matchIndexInOutput);
+        }
+      }
+
+
+      if (spanStartStack.Count > 0) {
+        throw new ArgumentException($"Saw {SpanStartString} without matching {SpanEndString}");
+      }
+
+      if (namedSpanStartStack.Count > 0) {
+        throw new ArgumentException($"Saw {NamedSpanEndString} without matching {NamedSpanEndString}");
```

Comment:
```suggestion
        throw new ArgumentException(@"Saw '{<' without matching '>}'");
```
Or you might want to create and use `NamedSpanStartString`